src/Microsoft.Extensions.Logging/Logger.cs

```diff
@@ -136,6 +139,21 @@ public IDisposable BeginScope<TState>(TState state)
             return scope;
         }
 
+        public IMetric DefineMetric(string name)
+        {
+            if (!_metrics.TryGetValue(name, out var metric))
```

Comment:
Is dictionary safe for multi thread read during the write?

Reply:
I'm honestly not sure. I was having the same concern and thinking I should just use ConcurrentDictionary